### PR TITLE
Upgrade lnd and Cancelable payment flag

### PR DESCRIPTION
**Pull request by hieblmi**
This PR upgrades to lnd to `v0.18.3-beta` and then adds `Cancelable` to the `SendPaymentRequest`.

### CODE REVIEW

File: router_client.go [hunk 1]
```diff
@@ -284,6 +284,13 @@ type SendPaymentRequest struct {
 
 	// AMP is set to true if the payment should be an AMP payment.
 	AMP bool
+
+	//  If set, the payment loop can be interrupted by manually canceling
```
Comment (guggero) [inline router_client.go:288]: nit: Godoc comments should start with the field name. Also, there are double spaces at the start of the comment.

**Review by guggero — COMMENTED** [review]
Looks good. I changed the base to add this to the `lnd-18-0` branch, which caused conflicts.

**Comment by guggero** [comment]
I think you need to revert the changes to `tools/`, otherwise you're downgrading the linter.

**Review by bhandras — APPROVED** [review]
LGTM, pending Oliver's comments 🎉

File: router_client.go [hunk 2]
```diff
@@ -284,6 +284,13 @@ type SendPaymentRequest struct {
 
 	// AMP is set to true if the payment should be an AMP payment.
 	AMP bool
+
+	//  Cancelable controls if the payment can be interrupted manually by
```
Comment (guggero) [inline router_client.go:288]: nit: double spaces still there...

**Review by guggero — APPROVED** [review]
LGTM :tada:

File: router_client.go (hunk 2 above)
Comment (hieblmi) [reply router_client.go]: fixed.

**Pull request merged**